Bio/PDB/internal_coords.py

```diff
@@ -0,0 +1,2765 @@
+              previous residue, 0 is this residue, 1 is next residue
+            - psi, phi, omg, omega, chi1, ...
+            - except for tuples, no option to access alternate disordered atoms
+
+        :return: Matching Hedron, Dihedron, or None.
+        """
+        rval = None
+        if isinstance(angle_key, tuple):
+            len_mkey = len(angle_key)
+            if 4 == len_mkey:
+                rval = self.dihedra.get(angle_key, None)
+            elif 3 == len_mkey:
+                rval = self.hedra.get(angle_key, None)
+        elif ":" in angle_key:
+            angle_key = self._get_ak_tuple(angle_key)
+            rval = self.dihedra.get(angle_key, None)
+        elif "psi" == angle_key:
+            if 0 == len(self.rnext):
+                return None
+            rn = self.rnext[0]
+            sN, sCA, sC = self.rak("N"), self.rak("CA"), self.rak("C")
+            nN = rn.rak("N")
+            rval = self.dihedra.get((sN, sCA, sC, nN), None)
+        elif "phi" == angle_key:
+            if 0 == len(self.rprev):
+                return None
+            rp = self.rprev[0]
+            pC, sN, sCA = rp.rak("C"), self.rak("N"), self.rak("CA")
+            sC = self.rak("C")
+            rval = rp.dihedra.get((pC, sN, sCA, sC), None)
+        elif "omg" == angle_key or "omega" == angle_key:
+            if 0 == len(self.rprev):
+                return None
+            rp = self.rprev[0]
+            pCA, pC, sN = rp.rak("CA"), rp.rak("C"), self.rak("N")
+            sCA = self.rak("CA")
+            rval = rp.dihedra.get((pCA, pC, sN, sCA), None)
+        elif angle_key.startswith("chi"):
+            sclist = ic_data_sidechains.get(self.lc, None)
+            if sclist is None:
+                return None
+            for akl in sclist:
+                if 5 == len(akl):
+                    if akl[4] == angle_key:
+                        klst = [self.rak(a) for a in akl[0:4]]
+                        rval = self.dihedra.get(tuple(klst), None)
+
+        return rval
+
+    def get_angle(self, angle_key):
+        """Get dihedron or hedron angle for specified key.
+
+        See pick_angle() for key specifications.
+        """
+        rval = self.pick_angle(angle_key)
+        if rval is not None:
+            return rval.get_angle()
+        return None
+
+    def set_angle(self, angle_key, v):
+        """Set dihedron or hedron angle for specified key.
+
+        See pick_angle() for key specifications.
+        """
+        rval = self.pick_angle(angle_key)
+        if rval is not None:
+            rval.set_angle(v)
+
+    def pick_length(self, ak_spec):
+        """Get list of hedra containing specified atom pair.
+
+        :param ak_spec:
+            - tuple of two AtomKeys
+            - string: two atom names separated by ':', e.g. 'N:CA' with
+              optional position specifier relative to self, e.g. '-1C:N' for
+              preceding peptide bond.
+
+        """
+      
```

Comment:
Maybe the `__repr__` method would be a better choice? I cannot print a dihedron:
```
Traceback (most recent call last):
  File "average_torsions.py", line 24, in <module>
    print(dihe)
TypeError: __repr__ returned non-string (type tuple)
```

Reply:
Could not work out the code you were commenting on, but changed __str__s to __repr__s which does seem a better choice.